Assets/Scripts/async-editor/async-editor.js

```diff
@@ -0,0 +1,216 @@
+        this.asyncEditorId = parameters.asyncEditorId;
+        this.providerName = parameters.providerName;
+        this.contentType = parameters.contentType;
+    }
+
+    loadEditor(contentId, editorGroup, callback) {
+        return fetch(this.createUrl(contentId, editorGroup))
+            .then((response) => response.json())
+            .then((data) => {
+                callback(true, data);
+            })
+            .catch((error) => {
+                callback(false, error);
+            });
+    }
+
+    submitEditor(contentId, editorGroup, nextEditorGroup, formData, callback) {
+        for (var pair of formData.entries()) {
+            console.log(pair[0]+ ', ' + pair[1]);
+        }
+        return fetch(this.createUrl(contentId, editorGroup, nextEditorGroup), {
+            method: 'post',
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
+            },
+            body: new URLSearchParams(formData) })
+            .then((response) => response.json())
+            .then((data) => {
+                callback(true, data);
+            })
+            .catch((error) => {
+                callback(false, error);
+            });
+    }
+
+    createUrl(contentId, editorGroup, nextEditorGroup) {
+        const url = new URL(this.apiUrl, document.baseURI);
+        const query = {
+            asyncEditorId: this.asyncEditorId,
+            providerName: this.providerName,
+            contentType: this.contentType,
+        };
+        if (contentId) query.contentId = contentId;
+        if (editorGroup) query.editorGroup = editorGroup;
+        if (nextEditorGroup) query.nextEditorGroup = nextEditorGroup;
+        url.search = new URLSearchParams(query).toString();
+
+        return url;
+    }
+}
+
+const router = new VueRouter();
+
+window.asyncEditor.editor = {
+    template: '#async-editor-template',
+    data() {
+        return {
+            asyncEditorId: '',
+            api: null,
+            message: '',
+            errorText: '',
+            contentId: '',
+            editorHtml: '',
+            validationSummaryHtml: '',
+            editorGroup: '',
+            editorGroups: [],
+            defaultErrorText: '',
+            scriptsHtml: ''
+        };
+    },
+    computed: {
+        progress(self) {
+            if (self.editorGroups.length < 1) return 0;
+            return self.editorGroups.filter(group => group.isFilled).length / self.editorGroups.length * 100;
+        },
+        showProgressBar(self) {
+            return self.editorGroups.length > 1;
+        }
+    },
+    watch: {
+        '$route.query': function(newVal, oldVal) {
+            this.processQuery();
+        }
+    },
+    router,
+    updated() {
+        const self = this;
+        if (self.scriptsHtml) {
+            self.scriptsHtml.match(/(?<=<script>).*?(?=<\/script>)/gms).forEach(function (match) {
+                window.eval(match);
+            });

```

Comment:
I mean don't use regex to parse HTML because it's unreliable. The DomParser should be used instead of the `self.scriptsHtml.match(/(?<=<script>).*?(?=<\/script>)/gms)`